WooCommerce/Classes/ViewRelated/Orders/Order Details/Address Edit/AddressFormViewModelProtocol.swift

```diff
@@ -401,6 +401,37 @@ open class AddressFormViewModel: ObservableObject {
         let secondaryEmailIsValid = secondaryFields.email.isEmpty || EmailFormatValidator.validate(string: fields.email)
         return primaryEmailIsValid && secondaryEmailIsValid
     }
+
+    /// Fills Order AddressFormFields with Customer details
+    ///
+    func customerSelectedFromSearch(customer: Customer) {
+        fillCustomerFields(customer: customer)
+        let addressesDiffer = customer.billing != customer.shipping
+        showDifferentAddressForm = addressesDiffer
+    }
+
+    private func fillCustomerFields(customer: Customer) {
+        fields = populate(fields: fields, with: customer, address: customer.billing)
+        secondaryFields = populate(fields: secondaryFields, with: customer, address: customer.shipping)
+    }
+
+    private func populate(fields: AddressFormFields, with customer: Customer, address: Address?) -> AddressFormFields {
+        var fields = fields
+
+        fields.firstName = customer.firstName ?? ""
+        fields.lastName = customer.lastName ?? ""
+        fields.email = customer.email
```

Comment:
That makes sense. I made the changes here: 121ba8a10f18fa7b0f78f998df81c82d6b186544

Something that peaks my interest with this change is that the email property is now an optional, but it can't actually be optional because is a mandatory field for shippable orders. This seems to happen because when we declare both `fields` and `secondaryFields` in the protocol, both inherit from `AddressFormFields`, however these have slightly different parameters in the API (for example shipping details does not have an email field)

> what happens if we save a different email address in each place? Does it get added to the order twice? Is that what Android does? Is it a bug, one way or the other?

This seems a bug from our end when declaring the model, as pointed above we're reusing the same `AddressFormFields` model for both shipping and billing, but these are not exactly the same in the API or WC core (it would seem that the API doc is outdated as well, as phone does not appear as property for shipping but in WC core it does). I've opened an issue here: https://github.com/woocommerce/woocommerce-ios/issues/7993 as most likely this can be improved.